### DISCUSSION TITLE
Is that auto-instrumentation javaagent working in android?

### DISCUSSION THREAD

**Discussion by sdes5317** [open-telemetry/opentelemetry-java#4404] [Q&A] [answered]
Hello i have tried to use it for android, and manual instrumentation work well, but i can't find any auto-instrumentation javaagent used by android.
is that working for android or has any example to use it?

**Comment by jkwatson** [comment, marked as answer]
Android doesn't support standard javaagent implementations, so no, this can't exist. There is some quasi-"automatic" instrumentation possible (see https://github.com/signalfx/splunk-otel-android for an example of such from Splunk), but it's nothing like what you would get from the standard opentelemetry javaagent for standard JVMs.

**Reply by sdes5317** [reply]
Thank you for your response,the answer just as what I expected,since I'm not a professional java developer I was not sure about it,
I have spent lot of time still can't prove it

**Comment by hhuynh123** [comment]
> Hello i have tried to use it for android, and manual instrumentation work well, but i can't find any auto-instrumentation javaagent used by android. is that working for android or has any example to use it?

Hi, can you provide example of manual instrumentation. I recently started looking into open telemetry for android. I am currently looking into splunk-otel-android to get more insight. Any information you can provide would be greatly appreciated.

**Reply by jack-berg** [reply]
Here are some links to examples:
- [open-telemetry/opentelemetry-java-docs](https://github.com/open-telemetry/opentelemetry-java-docs)
- [opentelemetry.io/docs/instrumentation/java/manual/](https://opentelemetry.io/docs/instrumentation/java/manual/)

**Reply by hhuynh123** [reply]
Thanks Jack. I am looking through those documents as well. I was trying to see if there was a basic example that i can tinker with
